Update Rails version in deprecation and add non-deprecated code

p8 opened the pull request:
Passing instance variables to `render` will be deprecated in the next
version, which is Rails 7.1.
Also add the non-deprecated implementation, which removes `@?` from the
Regexp, to make removal of the deprecation easier.

As noticed by @jhawthorn in #41464

rafaelfranca commented:
This looks incorrect. Some deprecations will be removed in 7.0, some in 7.1. I think all deprecations that say 7.0 are correct right now except the one you added.

rafaelfranca commented:
Better to just change yours, I'll revisit the numbers when I delete all the deprecated code that should be removed in 7.0

p8 commented:
Ok will fix it.

p8 committed:
Update Rails version in depraction and add non-deprecated code

Passing instance variables to `render` will be deprecated in the next
version, which is Rails 7.1.
Also add the non-deprecated implementation, which removes `@?` from the
Regexp, to make removal of the deprecation easier.

p8 commented:
@rafaelfranca I've updated the PR to only update my deprecation warning.
I've also added the non-deprecated version to make removal of the deprecation easier.

loust333 commented:
Hello guys, 

I'm having an issue concerning the deprecation. 

I'm trying to render a XML view and attach it to a model Object. I tried using locals but the show.builder does not recognize.

Currently working: 

`        xml = RtsStatementsController.new.render_to_string(action: "show", layout: false, locals: { :@company => @company, :@rts_statement => @rts_statement }, formats: [:xml])
`

Error (local not recognized)

`        xml = RtsStatementsController.new.render_to_string(action: "show", layout: false, locals: { company: company, rts_statement: rts_statement }, formats: [:xml])`



show.builder:

```
xml.Adresse do
    xml.AdressePhysique do
      xml.Numero company.number
      xml.Rue company.street
      xml.CodePostal company.postcode
      xml.Localite company.city
      xml.CodePays company.country
    end
  end

```
